pandas/core/indexes/extension.py

```diff
@@ -0,0 +1,65 @@
+"""
+Shared methods for Index subclasses backed by ExtensionArray.
+"""
+from pandas.util._decorators import cache_readonly
+
+
+def inherit_from_data(name, delegate, cache=False):
+    """
+    Make an alias for a method of the underlying ExtensionArray.
+
+    Parameters
+    ----------
+    name : str
+    delegate : class
+    cache : bool, default False
+        Whether to convert wrapped properties into cache_readonly
+
+    Returns
+    -------
+    method, property, or cache_readonly
+    """
+
+    attr = getattr(delegate, name)
+
+    if isinstance(attr, property):
+        if cache:
+            method = cache_readonly(attr.fget)
+
+        else:
+
+            def fget(self):
+                return getattr(self._data, name)
+
+            def fset(self, value):
+                setattr(self._data, name, value)
+
+            fget.__name__ = name
+            fget.__doc__ = attr.__doc__
+
+            method = property(fget, fset)
+
+    elif not callable(attr):
```

Comment:
Hmm somewhat confused by this function. Documented as working on `methods` but seems to accept attributes as well?

Reply:
Will update docstring.  This clause is for regular class attributes, in particular a few List[str] that we share